File: homework4_2/src/main/java/by/itacademy/homework4_2/MainActivity.java
package by.itacademy.homework4_2;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements LikeGoogleListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        LikeGoogle v = findViewById(R.id.likeGoogle);
        v.setListener(this::callback);
    }

    @Override
    public void callback(float x, float y) {
        Toast.makeText(MainActivity.this, String.format("You push x=%s, y=%s", x, y), Toast.LENGTH_SHORT).show();
    }
}

File: homework9/src/main/java/by/itacademy/homework9/MainActivity.kt
package by.itacademy.homework9

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import by.itacademy.homework9.databinding.ActivityMainBinding
import by.itacademy.homework9.model.HourlyWeatherModel
import by.itacademy.homework9.model.WeatherModel
import by.itacademy.homework9.presentation.HourlyWeatherAdapter
import by.itacademy.homework9.presentation.MainActivityListener
import by.itacademy.homework9.presentation.MainActivityPresenter
import by.itacademy.homework9.presentation.MainActivityPresenterImpl

class MainActivity : AppCompatActivity(), MainActivityListener {
    private lateinit var binding: ActivityMainBinding
    private val mainActivityPresenter: MainActivityPresenter by lazy { MainActivityPresenterImpl(this, baseContext) }
    private val weatherAdapter by lazy { HourlyWeatherAdapter() }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        with(binding) {
            recycler.apply {
                layoutManager = LinearLayoutManager(this@MainActivity)
                adapter = weatherAdapter
            }
            settingsButton.setOnClickListener { startActivity(Intent(this@MainActivity, SettingsActivity::class.java)) }
            cityButton.setOnClickListener { startActivity(Intent(this@MainActivity, CityActivity::class.java)) }
        }
    }

    override fun onStart() {
        super.onStart()
        mainActivityPresenter.getMainWeatherFromApi()
    }

    override fun getHourlyWeather(hourlyWeatherModel: List<HourlyWeatherModel>) {
        weatherAdapter.update(hourlyWeatherModel)
    }

    override fun getMainWeather(weatherModel: WeatherModel) {
        with(binding) {
            city.text = weatherModel.cityName
            degree.text = weatherModel.temp
            description.text = weatherModel.description
        }
    }

    override fun onError() {
        Toast.makeText(this, "ERROR", Toast.LENGTH_SHORT).show()
    }
}